src/main/java/wooteco/subway/line/LineController.java

```diff
@@ -0,0 +1,71 @@
+package wooteco.subway.line;
+
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import wooteco.subway.exception.LineNameDuplicatedException;
+import wooteco.subway.exception.LineNotFoundException;
+import wooteco.subway.line.dao.LineDao;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RestController
+public class LineController {
+
+    private final LineDao lineDao;
+
+    public LineController(LineDao lineDao) {
+        this.lineDao = lineDao;
+    }
+
+    @PostMapping("/lines")
+    public ResponseEntity<LineResponse> createLine(@RequestBody LineRequest lineRequest) {
+        final String name = lineRequest.getName();
+        final String color = lineRequest.getColor();
+        if (lineDao.findLineByName(name).isPresent()) {
+            return ResponseEntity.badRequest().build();
+        }
+        Line line = new Line(name, color);
+        Line createdLine = lineDao.save(line);
+        LineResponse lineResponse = new LineResponse(createdLine);
+        return ResponseEntity.created(URI.create("/lines/" + createdLine.getId()))
+            .body(lineResponse);
+    }
+
+    @GetMapping(value = "/lines", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<List<LineResponse>> showLines() {
+        List<Line> lines = lineDao.findAll();
+        List<LineResponse> lineResponses = lines.stream()
+            .map(LineResponse::new)
+            .collect(Collectors.toList());
+        return ResponseEntity.ok().body(lineResponses);
+    }
+
+    @GetMapping(value = "/lines/{id}")
+    public ResponseEntity<LineResponse> findLineById(@PathVariable Long id) {
+        Line line = lineDao.findLineById(id).orElseThrow(LineNotFoundException::new);
+        return ResponseEntity.ok(new LineResponse(line));
+    }
+
+    @PutMapping(value = "/lines/{id}")
+    public ResponseEntity<Void>updateLine(@PathVariable Long id, @RequestBody LineRequest lineRequest) {
+        lineDao.findLineById(id).orElseThrow(LineNotFoundException::new);
+
+        final Optional<Line> lineByName = lineDao.findLineByName(lineRequest.getName());
+        if (lineByName.isPresent() && lineByName.get().isNotSameId(id)) {
+            throw new LineNameDuplicatedException();
+        }
+
+        lineDao.update(id, lineRequest.getName(), lineRequest.getColor());
+        return ResponseEntity.ok().build();
+    }
+
+    @DeleteMapping("/lines/{id}")
+    public ResponseEntity<Void>removeLine(@PathVariable Long id) {
```

Comment:
```suggestion
    public ResponseEntity<Void> removeLine(@PathVariable Long id) {
```

존재하지 않는 id로 DELETE 요청을 보내도 204 응답이 오고 있는데 어떻게 생각하시나요?

Reply:
해당 부분은 없는걸 제거하기 때문에 에러가 발생하지는 않겠냐고 이야기가 나와서 이렇게 처리 했는데 이 경우도 예외 처리해 보도록 하겠습니다.